src/main/rx/jetiexbus.c

```diff
@@ -0,0 +1,595 @@
+    EXBUS_TRANS_RX,
+    EXBUS_TRANS_TX_READY,
+    EXBUS_TRANS_TX
+};
+
+enum exBusHeader_e {
+    EXBUS_HEADER_SYNC = 0,
+    EXBUS_HEADER_REQ,
+    EXBUS_HEADER_MSG_LEN,
+    EXBUS_HEADER_PACKET_ID,
+    EXBUS_HEADER_DATA_ID,
+    EXBUS_HEADER_SUBLEN,
+    EXBUS_HEADER_DATA
+};
+
+#ifdef TELEMETRY
+
+enum exTelHeader_e {
+    EXTEL_HEADER_SYNC = 0,
+    EXTEL_HEADER_TYPE_LEN,
+    EXTEL_HEADER_USN_LB,
+    EXTEL_HEADER_USN_HB,
+    EXTEL_HEADER_LSN_LB,
+    EXTEL_HEADER_LSN_HB,
+    EXTEL_HEADER_RES,
+    EXTEL_HEADER_ID,
+    EXTEL_HEADER_DATA
+};
+
+enum exDataType_e {
+    EX_TYPE_6b   = 0, // int6_t  Data type 6b (-31 ¸31)
+    EX_TYPE_14b  = 1, // int14_t Data type 14b (-8191 ¸8191)
+    EX_TYPE_22b  = 4, // int22_t Data type 22b (-2097151 ¸2097151)
+    EX_TYPE_DT   = 5, // int22_t Special data type – time and date
+    EX_TYPE_30b  = 8, // int30_t Data type 30b (-536870911 ¸536870911)
+    EX_TYPE_GPS  = 9  // int30_t Special data type – GPS coordinates:  lo/hi minute - lo/hi degree.
+};
+
+typedef struct exBusSensor_s{
+    const char *name;
+    const char *unit;
+    int32_t value;
+    const uint8_t exDataType;
+    const uint32_t decimals;
+} exBusSensor_t;
+
+
+#define SETMASK(decimals, bytes) (decimals << ((bytes*8)-3))
+#define RESETMASK(bytes) (~(3 << ((bytes*8)-3)))
+
+const uint32_t resetMask[]={ RESETMASK(1), RESETMASK(2), RESETMASK(3), RESETMASK(4) };
+
+// list of telemetry messages
+// after every 15 sensors a new header has to be inserted (e.g. "CF-Dev 1.12 S2")
+
+exBusSensor_t jetiExSensors[] = {
+    { "CF-Dev 1.12 S1", "",     0,      0,              0 },
+    { "Voltage",        "V",    0,      EX_TYPE_14b,    SETMASK(1,2) },
+    { "Current",        "A",    0,      EX_TYPE_14b,    SETMASK(2,2) },
+    { "Altitude",       "m",    0,      EX_TYPE_14b,    SETMASK(1,2) },
+    { "Capacity",       "mAh",  0,      EX_TYPE_14b,    SETMASK(0,2) }
+};
+
+enum exSensors_e {
+    EX_VOLTAGE = 1,
+    EX_CURRENT,
+    EX_ALTITUDE,
+    EX_CAPACITY
+};
+
+#define JETI_EX_SENSOR_COUNT (sizeof(jetiExSensors) / sizeof(exBusSensor_t))
+
+#endif //TELEMETRY
+
+static serialPort_t *jetiExBusPort;
+static serialPortConfig_t *portConfig;
+
+static uint8_t jetiExBusFrameState = EXBUS_STATE_ZERO;
+static uint8_t jetiExBusRequestState = EXBUS_STATE_ZERO;
+static uint8_t jetiExBusFramePosition;
+static uint8_t jetiExBusFrameLength;
+
+// Use max values for ram areas
+static uint8_t jetiExBusSerialFrame[JETIEXBUS_MAX_FRAME_SIZE+JETIEXBUS_MAX_REQUESTFRAME_SIZE];
+static uint8_t *jetiExBusFrame;
+
+
+static uint16_t jetiExBusChannelData[JETIEXBUS_CHANNEL_COUNT];
+static void jetiExBusDataReceive(uint16_t c);
+static uint16_t jetiExBusReadRawRC(rxRuntimeConfig_t *rxRuntimeConfig, uint8_t chan);
+
+
+#ifdef TELEMETRY
+
+static uint8_t jetiExBusTelemetryFrame[JETIEXBUS_MAX_FRAME_SIZE];
+static uint8_t jetiExBusTransceiveState = EXBUS_TRANS_RX;
+static uint8_t *jetiExTelemetry=&jetiExBusTelemetryFrame[EXBUS_HEADER_DATA];
```

Comment:
Sharing the buffer is OK, but it may be better to take the address before calling telemetry function. Or to define macro that transforms  jetiExBusTelemetryFrame to jetiExTelemetry.
Using another (static) variable for this aliasing  is confusing ..